knimin/handlers/ag_search.py

```diff
@@ -58,5 +60,7 @@ def post(self):
             display_results.append(login_display)
 
         # now render the page
-        self.render("ag_search.html", results=display_results,
-                    handouts=handouts, currentuser=self.current_user)
+        self.render("ag_search.html",
+                    results=xhtml_escape_recursive(display_results),
+                    handouts=map(xhtml_escape, handouts),
```

Comment:
cool. Looks like you are right and escaping is not necessary for rendering, but for unit testing.